wakuv2/waku.go

```diff
@@ -1707,6 +1707,10 @@ func (w *Waku) handleNetworkChangeFromApp(state connection.State) {
 
 func (w *Waku) ConnectionChanged(state connection.State) {
 	isOnline := !state.Offline
+	if isOnline && !w.onlineChecker.IsOnline() { // we are online and we were offline before
```

Comment:
Instead of the duplicated predicate logic and the comment, one idea is to create a predicate function, something like:

```diff
modified   wakuv2/waku.go
@@ -1705,9 +1705,13 @@ func (w *Waku) handleNetworkChangeFromApp(state connection.State) {
 	}
 }
 
+func (w *Waku) isGoingOnline(state connection.State) bool {
+	return !state.Offline && !w.onlineChecker.IsOnline()
+}
+
 func (w *Waku) ConnectionChanged(state connection.State) {
 	isOnline := !state.Offline
-	if isOnline && !w.onlineChecker.IsOnline() { // we are online and we were offline before
+	if w.isGoingOnline(state) {
 		//TODO: analyze if we need to discover and connect to peers for relay.
 		w.discoverAndConnectPeers()
 	}
@@ -1720,8 +1724,7 @@ func (w *Waku) ConnectionChanged(state connection.State) {
 		w.handleNetworkChangeFromApp(state)
 	} else {
 		// for lightClient state update and onlineChange is handled in filterManager.
-		// going online
-		if isOnline && !w.onlineChecker.IsOnline() {
+		if w.isGoingOnline(state) {
 			select {
 			case w.goingOnline <- struct{}{}:
 			default:
```